src/frontend/presenters/searchResultsPresenter.tsx

```diff
@@ -1,20 +1,29 @@
 import { useEffect, useState } from "react";
-import { canSearchImdb, searchImdb } from "../../backend/model/imdb";
+import { canSearchImdb, searchImdb, getTitleById } from "../../backend/model/imdb";
 import { useRecoilState, useRecoilValue, useSetRecoilState } from "recoil";
-import { selectedTitleAtom, searchValueState } from "../../backend/model/atoms";
-import { SearchResult, Title } from "../../backend/model/title";
+import { selectedTitleAtom, searchValueState, selectedTitle, selectedSeasonState } from "../../backend/model/atoms";
+import { SearchResult, Title, TitleId } from "../../backend/model/title";
 
 import SearchResultsView from "../views/searchResultsView";
 
 export default function SearchResults() {
 	const [titles, setTitles] = useState(null as SearchResult[] | null);
 	const searchValue = useRecoilValue(searchValueState);
-	const setSelectedTitle = useSetRecoilState(selectedTitleAtom);
+	const setSelectedTitleId = useSetRecoilState(selectedTitleAtom);
+	const setSelectedTitle = useSetRecoilState(selectedTitle);
+	const setSelectedSeason = useSetRecoilState(selectedSeasonState);
 
 	useEffect(() => {
 		setTitles(null);
+		setSelectedSeason("");
 		searchImdb(searchValue, false).then(t => setTitles(t)).catch((e: Error) => window.alert("Search failed: " + e.message));
 	}, [searchValue]);
 
-	return <SearchResultsView loading={!titles} titles={titles || []} onSelectTitle={t => setSelectedTitle(t.id)}></SearchResultsView>;
+	function setTitle(id: TitleId){
+		setSelectedTitleId(id);

```

Comment:
We should reset selectedTitle here too, otherwise the old title shows up in details view for a few seconds.

Reply:
Legit! Will add that!